java/vector/src/test/java/org/apache/arrow/vector/TestLargeListVector.java

```diff
@@ -723,23 +723,23 @@ public void testGetBufferAddress() throws Exception {
       Object result = listVector.getObject(0);
       ArrayList<Long> resultSet = (ArrayList<Long>) result;
       assertEquals(3, resultSet.size());
-      assertEquals(new Long(50), resultSet.get(0));
-      assertEquals(new Long(100), resultSet.get(1));
-      assertEquals(new Long(200), resultSet.get(2));
+      assertEquals(Long.valueOf(50), resultSet.get(0));
+      assertEquals(Long.valueOf(100), resultSet.get(1));
+      assertEquals(Long.valueOf(200), resultSet.get(2));
```

Comment:
weird; wouldn't 200L be ok here?

Reply:
Ah I couldn't agree more, but I honestly I was naively making the changes based on the suggestion in the warning. Sorry about that. Let me reiterate this step.

Reply:
Also thinking about it

`10L`  is `long` and `resultSet.get(i)` is `Long`

Reply:
`long actual = dataVector.getObject(offset);` 
We should probably change this and then do the rest?

Reply:
Shall we do it like this?

Reply:
No, I think it's fine to use 10L and rely on autoboxing. And since resultSet.get may return null it's not valid to assume it'll be non-null.

Reply:
There is a problem if we just do `assertEquals(200L, actual)`

```bash
error: reference to assertEquals is ambiguous
  both method assertEquals(long,long) in Assert and method assertEquals(Object,Object) in Assert match
```
won't compile.

Reply:
Aha...In that case valueOf like you have is OK then. Thanks.